jina/types/document/uid.py

```diff
@@ -28,13 +28,21 @@
 from ...helper import typename
 from ...proto.jina_pb2 import DocumentProto
 
-_doc_field_mask = None
+from jina.logging import default_logger
+
 _digest_size = 8
 _id_regex = re.compile(r'[0-9a-fA-F]{16}')
+_warned_deprecation = False
+
 
+def get_content_hash(doc: 'DocumentProto') -> str:
+    """ Generate a new hexdigest based on the content of the document.
 
-def new_doc_hash(doc: 'DocumentProto') -> int:
-    return id2hash(new_doc_id(doc))
+    :param doc: a non-empty document
+    :return: the hexdigest based on :meth:`blake2b`
+    """
+    # TODO: once `new_doc_id` is removed, the content of this function can directly move to the `Document`.
+    return blake2b(doc.SerializeToString(), digest_size=_digest_size).hexdigest()
```

Comment:
Shouldn't the content hash not depend on the id?

Reply:
Fair enough. That's where the `_doc_field_mask` comes into play. I'll add that.

Reply:
It should depend on the document content only. The ID needs to be totally independent.

Reply:
Ah it was the other way around...